Tasks/Common/webdeployment-common-v2/npmdomutility.ts

```diff
@@ -0,0 +1,98 @@
+var varUtility = require("./variableutility.js");
+var DOMParser = require('xmldom').DOMParser;
+
+export class NpmDomUtility  {
+
+    private xmlDomLookUpTable = {};
+    private headerContent;
+    private xmlDom;
+
+    public constructor(xmlContent) {
+        this.xmlDomLookUpTable = {};
+        this.headerContent = null;
+        this.xmlDom = new DOMParser().parseFromString(xmlContent,"text/xml");
+        this.readHeader(xmlContent);
+        this.buildLookUpTable(this.xmlDom);
+    }
+
+    public getXmlDom() {
+        return this.xmlDom;
+    }
+
+    private readHeader(xmlContent) {
+        var index = xmlContent.indexOf('\n');
+        if(index > -1) {
+            var firstLine = xmlContent.substring(0,index).trim();
+            if(firstLine.startsWith("<?") && firstLine.endsWith("?>")) {
+                this.headerContent = firstLine; 
+            }
+        }
+    }
+
+    public getContentWithHeader(xmlDom) {
+        return xmlDom ? xmlDom.toString() : "";
+    }
+
+    /**
+     * Define method to create a lookup for DOM 
+     */
+    private buildLookUpTable(node) {
+        if(node){
+            var nodeName = node.nodeName;
+            if(nodeName){
+                nodeName = nodeName.toLowerCase();
+                var listOfNodes = this.xmlDomLookUpTable[nodeName];
+                if(listOfNodes == null || !(Array.isArray(listOfNodes))) {
+                    listOfNodes = [];
+                    this.xmlDomLookUpTable[nodeName] = listOfNodes;
```

Comment:
Shouldn't this be done after the node is pushed in listnodes or node should be pushed directly in this.xmlDomLookUpTable[nodeName]. The logic is a bit confusing here